PR #329 breaks -flto unless using KOS_INIT_FLAGS(INIT_NET)

For example, when compiling `pvrmark`:

```
$ make
rm -f pvrmark.elf
kos-cc  -c pvrmark.c -o pvrmark.o
kos-cc -o pvrmark.elf pvrmark.o
/opt/toolchains/dc/sh-elf-13.2.0/bin/../lib/gcc/sh-elf/13.2.0/../../../../sh-elf/bin/ld: /tmp/ccA5SDhk.ltrans1.ltrans.o: in function `_dcls_recv_loop':
/opt/toolchains/dc/kos/examples/dreamcast/pvr/pvrmark/../../../../kernel/arch/dreamcast/fs/fs_dclsocket.c:118:(.text.dcls_recv_loop+0x19c): undefined reference to `_abort'
collect2: error: ld returned 1 exit status
make: *** [Makefile:23: pvrmark.elf] Error 1
```

Reverting PR #329 fixes the issue.

--- a/kernel/libc/koslib/abort.c
+++ b/kernel/libc/koslib/abort.c
@@ -9,7 +9,7 @@
 #include <arch/arch.h>
 
 /* This is probably the closest mapping we've got for abort() */
-void abort(void) {
+__attribute__((used)) void abort(void) {
     arch_exit();
 }
 

